client/app/controllers/shows.controller.js

```diff
@@ -1,22 +1,29 @@
 'use strict';
 
 angular.module('Tour-Track')
-.controller('ShowsCtrl', ['$scope', 'allShows', 'ShowFactory', 'VenueFactory', '$timeout', function($scope, allShows, ShowFactory, VenueFactory, $timeout) {
+.controller('ShowsCtrl', ['$scope', 'allShows', 'ShowFactory', '$timeout', function($scope, allShows, ShowFactory, $timeout) {
+
+  $scope.allShows = allShows;
 
   var years = [];
-  $scope.filter = { date: "", venue: "", location: "" };
+  $scope.filter = { year: "", date: "", venue: "", location: "" };
```

Comment:
filter is bad name here. We don't want to assign properties or variables to reserved words(filter, map, each, function, array, object, etc,)